File: ios/StillMoment/Presentation/Views/GuidedMeditations/TrimEditor/TrimReadoutCards.swift
//
//  TrimReadoutCards.swift
//  Still Moment
//
//  Presentation Layer — Anfang/Ende readout cards for the trim editor (shared-107).
//

import SwiftUI

/// The two "Anfang"/"Ende" cards below the waveform. Tapping a card selects the
/// corresponding point as active; the active card is highlighted. Exposed as buttons
/// with the `.isSelected` trait for the active one.
struct TrimReadoutCards: View {
    // MARK: Internal

    let start: TimeInterval
    let end: TimeInterval
    let activePoint: TrimPoint
    let onSelect: (TrimPoint) -> Void

    var body: some View {
        HStack(spacing: 10) {
            self.card(
                point: .start,
                label: "trim_editor.card.start",
                value: self.start
            )
            self.card(
                point: .end,
                label: "trim_editor.card.end",
                value: self.end
            )
        }
    }

    // MARK: Private

    @Environment(\.themeColors)
    private var theme

    private func card(point: TrimPoint, label: LocalizedStringKey, value: TimeInterval) -> some View {
        let isActive = self.activePoint == point
        let labelKey = point == .start ? "trim_editor.card.start" : "trim_editor.card.end"
        return Button {
            self.onSelect(point)
        } label: {
            VStack(alignment: .leading, spacing: 4) {
                Text(label)
                    .textStyle(.eyebrow, color: \.textSecondary)
                Text(EditSheetState.formatTime(value))
                    .textStyle(.title, monospacedDigits: true)
                    .foregroundColor(isActive ? self.theme.interactive : self.theme.textPrimary)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.horizontal, 14)
            .padding(.vertical, 10)
            .background(
                RoundedRectangle(cornerRadius: 14)
                    .fill(isActive ? self.theme.accentBubbleBackground : self.theme.cardBackground)
            )
            .overlay(
                RoundedRectangle(cornerRadius: 14)
                    .strokeBorder(
                        isActive ? self.theme.interactive.opacity(0.4) : self.theme.cardBorder,
                        lineWidth: 1
                    )
            )
        }
        .buttonStyle(.plain)
        .accessibilityElement(children: .ignore)
        .accessibilityLabel(Text(NSLocalizedString(labelKey, comment: "Trim point card label")))
        .accessibilityValue(Text(EditSheetState.formatTime(value)))
        .accessibilityHint(Text("trim_editor.a11y.cardHint"))
        .accessibilityAddTraits(isActive ? [.isSelected] : [])
    }
}

File: android/app/src/main/kotlin/com/stillmoment/presentation/ui/meditations/trimeditor/TrimReadoutCards.kt
package com.stillmoment.presentation.ui.meditations.trimeditor

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ZoomIn
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.stillmoment.R
import com.stillmoment.domain.models.TrimPoint
import com.stillmoment.presentation.ui.theme.LocalStillMomentColors
import com.stillmoment.presentation.ui.theme.TextStyle
import com.stillmoment.presentation.ui.theme.toComposeTextStyle

/**
 * The two "Anfang"/"Ende" cards below the waveform (shared-107/108). Tapping a card selects
 * the corresponding point as active and zooms the track onto it — a small magnifier icon
 * signals the zoom affordance. The active card is highlighted. 1:1 port of iOS `TrimReadoutCards`.
 */
@Composable
fun TrimReadoutCards(
    startMs: Long,
    endMs: Long,
    activePoint: TrimPoint,
    onSelect: (TrimPoint) -> Unit,
    modifier: Modifier = Modifier
) {
    Row(modifier = modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
        ReadoutCard(
            point = TrimPoint.START,
            labelRes = R.string.trim_editor_card_start,
            valueMs = startMs,
            isActive = activePoint == TrimPoint.START,
            onSelect = onSelect,
            modifier = Modifier.weight(1f)
        )
        ReadoutCard(
            point = TrimPoint.END,
            labelRes = R.string.trim_editor_card_end,
            valueMs = endMs,
            isActive = activePoint == TrimPoint.END,
            onSelect = onSelect,
            modifier = Modifier.weight(1f)
        )
    }
}

@Composable
private fun ReadoutCard(
    point: TrimPoint,
    labelRes: Int,
    valueMs: Long,
    isActive: Boolean,
    onSelect: (TrimPoint) -> Unit,
    modifier: Modifier = Modifier
) {
    val theme = LocalStillMomentColors.current
    val shape = RoundedCornerShape(14.dp)
    val label = stringResource(labelRes)
    val hint = stringResource(R.string.trim_editor_a11y_card_hint)
    Column(
        modifier = modifier
            .clip(shape)
            .background(if (isActive) theme.accentBubbleBackground else theme.cardBackground)
            .border(
                width = 1.dp,
                color = if (isActive) theme.interactive.copy(alpha = 0.4f) else theme.cardBorder,
                shape = shape
            )
            .clickable { onSelect(point) }
            .padding(horizontal = 14.dp, vertical = 10.dp)
            .semantics { contentDescription = "$label, ${formatTrimTime(valueMs)}. $hint" },
        verticalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text(
                text = TextStyle.eyebrow.applyCase(label),
                style = TextStyle.eyebrow.toComposeTextStyle(),
                color = theme.textPrimary.copy(alpha = 0.6f),
                modifier = Modifier.weight(1f)
            )
            Icon(
                Icons.Filled.ZoomIn,
                contentDescription = null,
                tint = if (isActive) theme.interactive else theme.textPrimary.copy(alpha = 0.5f)
            )
        }
        Text(
            text = formatTrimTime(valueMs),
            style = TextStyle.title.toComposeTextStyle(),
            color = if (isActive) theme.interactive else theme.textPrimary
        )
    }
}
